glib/src/closure.rs

```diff
@@ -27,17 +30,139 @@ wrapper! {
     }
 }
 
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+pub struct RustClosure(Closure);
+
+impl RustClosure {
+    /// Creates a new closure around a Rust closure.
+    ///
+    /// See [`glib::closure!`] for a way to create a closure with concrete types.
+    ///
+    /// # Panics
+    ///
+    /// Invoking the closure with wrong argument types or returning the wrong return value type
+    /// will panic.
+    #[doc(alias = "g_closure_new")]
```

Comment:
Would be nice to have a code example (run when using `cargo test`) for each of these methods.

Reply:
We tests covering them already. Ideally nobody should use these functions unless they already know what they're doing. There's the macro for the other case.

Reply:
Code examples don't hurt. Even if you know what you're doing.

Reply:
I forward all people using the API wrongly to you then :)

I'll copy&paste one of the tests into the docs later.

Reply:
That's another debate.

Reply:
There is a code example now on the two constructors.